ts/input/tex/TexParser.ts

```diff
@@ -183,11 +183,9 @@ export default class TexParser {
     let c: string;
     let n: number;
     while (this.i < this.string.length) {
-      c = this.string.charAt(this.i++);
-      n = c.charCodeAt(0);
-      if (n >= 0xD800 && n < 0xDC00) {
-        c += this.string.charAt(this.i++);
-      }
+      n = this.string.codePointAt(this.i);
+      c = String.fromCodePoint(n);
```

Comment:
Collapse into one and add a utility method for
`String.fromCodePoint(this.string.codePointAt`

Reply:
Done (and for the ones below)